src/sinks/elasticsearch/config.rs

```diff
@@ -368,44 +379,63 @@ impl DataStreamConfig {
 #[async_trait::async_trait]
 impl SinkConfig for ElasticsearchConfig {
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
-        let common = ElasticsearchCommon::parse_config(self).await?;
+        let commons = ElasticsearchCommon::parse_many(self).await?;
+        let common = commons[0].clone();
 
-        let http_client = HttpClient::new(common.tls_settings.clone(), cx.proxy())?;
-        let batch_settings = self.batch.into_batcher_settings()?;
+        let client = HttpClient::new(common.tls_settings.clone(), cx.proxy())?;
 
         let request_limits = self
             .request
             .tower
             .unwrap_with(&TowerRequestConfig::default());
 
-        let http_request_builder = HttpRequestBuilder {
-            bulk_uri: common.bulk_uri.clone(),
-            http_request_config: self.request.clone(),
-            http_auth: common.http_auth.clone(),
-            query_params: common.query_params.clone(),
-            region: common.region.clone(),
-            compression: self.compression,
-            credentials_provider: common.aws_auth.clone(),
-        };
+        let stream = if commons.len() > 1 {
```

Comment:
Would it be possible to just use the distributed service scaffolding even if there is only 1 endpoint rather than having two separate paths here?

Reply:
It would change current behavior for a single endpoint, although exclusively in the domain of retries. Other than that, it would indeed be simpler.

Reply:
How would the retry behavior differ?

Reply:
It would add circuit breaker layer/behavior that would trigger on multiple consecutive errors so it will be more sensitive to errors. It's difficult to say what exact difference would it be, but I think it will take more time to start sending again in some cases. For multiple endpoints that doesn't matter so much since there are alternatives.

Reply:
Changed.